### PR TITLE
Add projinfo option --list-crs (supports --area)

### PR DESCRIPTION
**Pull request by jjimenezshaw**
- [x] Tests added
 - [x] Added clear title that can be used to generate release notes
 - [x] Fully documented, including updating `docs/source/*.rst` for new API

Variation of #2655. (Same behavior, different implementation).

It also uses the option `--area`. For that it creates an `Extent` from `--bbox` or `--area` (in the same way as other parts of `projinfo`. New function `makeBboxFilter` for that purpose) and intersects every CRS listed. The behavior is driven by `--spatial-test`

It does not use the C API, but directly the C++ function `getCRSInfoList()`.

### CODE REVIEW

File: src/apps/projinfo.cpp [hunk 1]
```diff
@@ -1248,6 +1332,77 @@ int main(int argc, char **argv) {
...
+                auto list = factory->getCRSInfoList();
+                for (const auto &info : list) {
+                    if (!allow_deprecated && info.deprecated) {
+                        continue;
+                    }
+                    if (!types.empty() && types.find(info.type) == types.end()) {
+                        continue;
+                    }
+                    if (bboxFilter) {
+                        auto crsExtent = Extent::createFromBBOX(
```
Comment (rouault) [inline src/apps/projinfo.cpp:1392]: you should check info.bbox_valid

File: src/apps/projinfo.cpp [hunk 2]
```diff
@@ -1248,6 +1332,77 @@ int main(int argc, char **argv) {
...
+                            info.west_lon_degree, info.south_lat_degree,
+                            info.east_lon_degree, info.north_lat_degree);
+                        if (!bboxFilter->intersects(crsExtent)) {
```
Comment (rouault) [inline src/apps/projinfo.cpp]: I'm wondering if we shouldn't take the --spatial-test contains|intersects option into account . The default is SpatialCriterion::STRICT_CONTAINMENT that is we would only list CRS whose area of use is entirely contained in the specified bbox (thus bboxFilter->contains(crsExtent) . And if using --spatial-test intersects, what you do currently. Or if you prefer the current behaviour, in --list-crs mode change the default value to be intersects, but allow the override of --spatial-test contains

Comment (jjimenezshaw) [reply src/apps/projinfo.cpp]: It makes sense. Coming in next commit.

File: src/apps/projinfo.cpp (hunk 1 above)
Comment (jjimenezshaw) [reply src/apps/projinfo.cpp:1392]: So the logic should be like this?:
```if (bboxFilter && !info.bbox_valid) { continue; }```
(that means that CRS without extent definition are not listed if `--bbox` or `--area` are used)

Comment (rouault) [reply src/apps/projinfo.cpp:1392]: sounds reasonable. That's more an hypothetical sense since a well behaved database should include extent for all CRS

**Pull request merged**